### DISCUSSION TITLE
Using "fullscreen" action and move a window between different sized desktops always keeps it fullscreen

### DISCUSSION THREAD

**Discussion by mfn** [rxhanson/Rectangle#1104] [Q&A] [answered]
Ok, hear me out 😅

I just switched to Rectangle (because Apple Silicon) and that way I learned that Spectacle was abandon 😿

Anyhow, Rectangle is a perfect replacement, except this one thing:
- two FullHD 1920x1080 monitors to the left of my MacBook
- A window in fullscreen on the laptop screen is actually _smaller_, because the resolution I use on the laptop is 1496x967 (though it renders as 2992x1934). Basically "that middle option" in my display setting:
  ![image](https://user-images.githubusercontent.com/87493/224023029-87db6b30-1b08-403b-a9c9-ddae39a869cb.png)

With Rectangle, when I move the window to the left whose desktop has a bigger resolution, the window keeps its "fullscreen" property.

With Spectacle, when I move the window to the left whose desktop has a bigger resolution, the window keeps its "current size as if it still where on the laptop screen".

I'm looking for a way to keep the way Spectacle behavior.

Ok, screenshots:
- laptop screen: firefox in "fullscreen" 
  <img width="640" src="https://user-images.githubusercontent.com/87493/224025214-b5287820-f488-4354-a9be-6cc35c3bbf54.png">
- left monitor, moved with Rectangle: keeps "fullscreen"
  <img width="640" src="https://user-images.githubusercontent.com/87493/224025413-b0d3758b-d824-4104-bae2-daf7cfd47004.png">
- left monitor, moved with Spectacle: keeps "window size"
  <img width="640" src="https://user-images.githubusercontent.com/87493/224025514-dac8f7ed-b7fe-4a85-8da3-ff7e077d2f72.png">

Rectangle settings:
![image](https://user-images.githubusercontent.com/87493/224025801-45660c6a-f530-4ec2-b53e-fe47ea21346b.png)

Spectacle settings:
![image](https://user-images.githubusercontent.com/87493/224025710-aabeef13-9b0c-421f-a239-1219002fa5f5.png)

Is there a way, or setting I maybe missed, to get the "spectacle behaviour"?

thanks 🙏🏼

**Comment by rxhanson** [comment, marked as answer]
I think what you're looking for is the behavior from executing this terminal command:

`defaults write com.knollsoft.Rectangle autoMaximize -int 2`

Then restart Rectangle.

Let me know if that's not it.

**Comment by mfn** [comment]
# YES 🙏🏼
